warehouse/query-core/src/main/java/datawave/query/ranges/RangeFactory.java

```diff
@@ -0,0 +1,87 @@
+package datawave.query.ranges;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+
+/**
+ * This class provides a standard way of building ranges with clear intent.
+ */
+public class RangeFactory {
+    
+    public static final String NULL_BYTE_STRING = "\u0000";
+    public static final String MAX_UNICODE_STRING = new String(Character.toChars(Character.MAX_CODE_POINT));
+    private static final String FIRST_SHARD = "_0";
+    
+    /**
+     * Builds a tld document range that can be passed to the {@link datawave.query.iterator.QueryIterator}
```

Comment:
Copy/paste error I think. This one does not build a TLD document range.

Reply:
Fixed in next commit